RetailApp/RetailApp.Data/ConfigOptions/RepositoryOptions.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace RetailApp.Data.ConfigOptions
+{
+    public class RepositoryOptions
```

Comment:
Remove it. You cannot use it with appsettings json settings. It should be dynamically passed (DbContextTypes) via the constructors.